packages/file-graph/fileGraph.js

```diff
@@ -1,26 +1,72 @@
 import FileNode from './fileNode.js'
+import { FileNotExistsError } from './errors.js'
 
 export default class FileGraph {
   constructor(root) {
     this.root = new FileNode(root)
   }
 
-  add(file) {
+  add(files) {
     // Logic to add the file node to the graph
     // You can implement the logic to find the appropriate location in the file graph
     // If the file node already exists, throw an error
+
+    files.forEach((file) => {
+      this.root.addChild(file, { makeParents: true })
+    })
   }
 
-  remove(file) {
+  remove(files) {
     // Logic to remove a file node from the graph
     // If the file node is a folder, also remove all children
     // If the file node doesn't exist, throw an error
+
+    files.forEach((file) => {
+      const pathFolders = file.path
+        .split('/')
+        .filter((folder) => folder.length > 0)
+      const fileName = pathFolders.pop()
+      let currentNode = this.root
+
+      for (const folder of pathFolders) {
+        currentNode = currentNode.children.find(
+          (child) => child.name === folder
+        )
+        if (!currentNode) {
+          throw new FileNotExistsError(
+            `File not found: ${fileToFind.path}, ${this.toString()}`
+          )
+        }
+      }
+
+      if (!currentNode.children.some((child) => child.name === fileName)) {
+        throw new FileNotExistsError(
+          `File not found: ${file.path}, ${this.toString()}`
+        )
+      } else {
+        currentNode.children = currentNode.children.filter(
+          (child) => child.name !== fileName
+        )
+      }
+    })
   }
 
   discover(eventFiles) {
     // Logic to handle the discover event
     // Overwrite the file graph with the provided files
     // Remove file nodes that don't exist anymore and add new ones
+
+    const newRoot = new FileNode({
+      path: '/',
+      type: 'folder',
+      discovered: true,
+    })
```

Comment:
In my design, the eventFiles is generated by 'ls' command, so it would consist mainly files in one folder (until user ls two folder at the same time.) Due to how FileNode.merge's implementation, it would remove the files not exists in the tree, so if you create newRoot on '/', it would remove more file than what we want. Instead, you should create root from the first item of the eventFiles. Please refer to the merging test in fileNode.test.js